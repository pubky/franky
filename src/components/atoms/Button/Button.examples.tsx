'use client';

import * as React from 'react';

import { Button } from './Button';
import type { ButtonVariant, ButtonSize } from './Button.types';

export function ButtonExamples() {
  const variants: ButtonVariant[] = [
    'default',
    'destructive',
    'outline',
    'secondary',
    'ghost',
    'brand',
    'link',
    'dark',
    'dark-outline',
  ];

  const sizes: ButtonSize[] = ['sm', 'default', 'lg', 'icon'];

  return (
    <div className="p-8 space-y-12 bg-background min-h-screen">
      <div className="max-w-6xl mx-auto">
        <h1 className="text-4xl font-bold mb-2">Button Examples</h1>
        <p className="text-muted-foreground mb-8">
          Comprehensive showcase of all Button component variations and use cases.
        </p>

        {/* All Variants */}
        <section className="space-y-4">
          <h2 className="text-2xl font-semibold">Variants</h2>
          <div className="grid grid-cols-2 md:grid-cols-3 lg:grid-cols-5 gap-4">
            {variants.map((variant) => (
              <div key={variant} className="space-y-2">
                <Button variant={variant} className="w-full">
                  {variant}
                </Button>
                <code className="text-xs text-muted-foreground block text-center">variant=&quot;{variant}&quot;</code>
              </div>
            ))}
          </div>
        </section>

        {/* All Sizes */}
        <section className="space-y-4">
          <h2 className="text-2xl font-semibold">Sizes</h2>
          <div className="flex items-center gap-4 flex-wrap">
            {sizes.map((size) => (
              <div key={size} className="space-y-2 text-center">
                <Button size={size}>{size === 'icon' ? 'üîç' : `Size ${size}`}</Button>
                <code className="text-xs text-muted-foreground block">size=&quot;{size}&quot;</code>
              </div>
            ))}
          </div>
        </section>

        {/* Size x Variant Matrix */}
        <section className="space-y-4">
          <h2 className="text-2xl font-semibold">Size √ó Variant Matrix</h2>
          <div className="overflow-x-auto">
            <table className="w-full border-collapse">
              <thead>
                <tr>
                  <th className="text-left p-2 border border-border">Variant</th>
                  {sizes.map((size) => (
                    <th key={size} className="text-center p-2 border border-border min-w-24">
                      {size}
                    </th>
                  ))}
                </tr>
              </thead>
              <tbody>
                {variants.map((variant) => (
                  <tr key={variant}>
                    <td className="p-2 border border-border font-mono text-sm">{variant}</td>
                    {sizes.map((size) => (
                      <td key={`${variant}-${size}`} className="p-2 border border-border text-center">
                        <Button variant={variant} size={size}>
                          {size === 'icon' ? '‚úì' : 'Button'}
                        </Button>
                      </td>
                    ))}
                  </tr>
                ))}
              </tbody>
            </table>
          </div>
        </section>

        {/* States */}
        <section className="space-y-4">
          <h2 className="text-2xl font-semibold">States</h2>
          <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6">
            <div className="space-y-2">
              <h3 className="font-medium">Normal</h3>
              <Button>Normal Button</Button>
            </div>
            <div className="space-y-2">
              <h3 className="font-medium">Disabled</h3>
              <Button disabled>Disabled Button</Button>
            </div>
            <div className="space-y-2">
              <h3 className="font-medium">Loading State</h3>
              <Button disabled className="opacity-75">
                <span className="animate-spin">‚è≥</span> Loading...
              </Button>
            </div>
            <div className="space-y-2">
              <h3 className="font-medium">With Icon</h3>
              <Button>
                <span>üìÑ</span> With Icon
              </Button>
            </div>
          </div>
        </section>

        {/* AsChild Usage */}
        <section className="space-y-4">
          <h2 className="text-2xl font-semibold">AsChild Usage</h2>
          <div className="space-y-4">
            <div className="space-y-2">
              <h3 className="font-medium">Button as Link</h3>
              <Button asChild variant="outline">
                <a href="#" className="inline-flex">
                  External Link üîó
                </a>
              </Button>
              <code className="text-xs text-muted-foreground block">
                &lt;Button asChild&gt;&lt;a href=&quot;#&quot;&gt;Link&lt;/a&gt;&lt;/Button&gt;
              </code>
            </div>
            <div className="space-y-2">
              <h3 className="font-medium">Button as Div</h3>
              <Button asChild variant="secondary">
                <div>Custom Div Element</div>
              </Button>
              <code className="text-xs text-muted-foreground block">
                &lt;Button asChild&gt;&lt;div&gt;Content&lt;/div&gt;&lt;/Button&gt;
              </code>
            </div>
          </div>
        </section>

        {/* Interactive Examples */}
        <section className="space-y-4">
          <h2 className="text-2xl font-semibold">Interactive Examples</h2>
          <div className="grid grid-cols-1 md:grid-cols-2 gap-6">
            <div className="space-y-3">
              <h3 className="font-medium">Action Buttons</h3>
              <div className="flex gap-2 flex-wrap">
                <Button variant="default" onClick={() => alert('Default clicked!')}>
                  Primary Action
                </Button>
                <Button variant="secondary" onClick={() => alert('Secondary clicked!')}>
                  Secondary
                </Button>
                <Button variant="outline" onClick={() => alert('Outline clicked!')}>
                  Tertiary
                </Button>
              </div>
            </div>
            <div className="space-y-3">
              <h3 className="font-medium">Danger Zone</h3>
              <div className="flex gap-2 flex-wrap">
                <Button
                  variant="destructive"
                  onClick={() => confirm('Are you sure you want to delete?') && alert('Deleted!')}
                >
                  Delete
                </Button>
                <Button variant="outline" onClick={() => alert('Cancelled!')}>
                  Cancel
                </Button>
              </div>
            </div>
          </div>
        </section>

        {/* Form Examples */}
        <section className="space-y-4">
          <h2 className="text-2xl font-semibold">Form Usage</h2>
          <div className="max-w-md space-y-4 p-6 border border-border rounded-lg">
            <div>
              <label className="block text-sm font-medium mb-2">Email</label>
              <input type="email" className="w-full p-2 border border-border rounded-md" placeholder="your@email.com" />
            </div>
            <div className="flex gap-2">
              <Button type="submit" className="flex-1">
                Submit Form
              </Button>
              <Button type="button" variant="outline">
                Reset
              </Button>
            </div>
          </div>
        </section>

        {/* Code Examples */}
        <section className="space-y-4">
          <h2 className="text-2xl font-semibold">Code Examples</h2>
          <div className="grid grid-cols-1 lg:grid-cols-2 gap-6">
            <div className="space-y-2">
              <h3 className="font-medium">Basic Usage</h3>
              <pre className="bg-muted p-4 rounded-md text-sm overflow-x-auto">
                {`// Basic button
<Button>Click me</Button>

// With variant
<Button variant="destructive">
  Delete
</Button>

// With size
<Button size="lg">
  Large Button
</Button>

// Combined
<Button variant="outline" size="sm">
  Small Outline
</Button>`}
              </pre>
            </div>
            <div className="space-y-2">
              <h3 className="font-medium">Advanced Usage</h3>
              <pre className="bg-muted p-4 rounded-md text-sm overflow-x-auto">
                {`// As child element
<Button asChild>
  <Link href="/profile">
    Go to Profile
  </Link>
</Button>

// With custom styling
<Button 
  variant="ghost" 
  className="hover:bg-red-100"
>
  Custom Button
</Button>

// Event handlers
<Button 
  onClick={handleClick}
  disabled={isLoading}
>
  {isLoading ? 'Loading...' : 'Submit'}
</Button>`}
              </pre>
            </div>
          </div>
        </section>

        {/* Best Practices */}
        <section className="space-y-4">
          <h2 className="text-2xl font-semibold">Best Practices</h2>
          <div className="grid grid-cols-1 md:grid-cols-2 gap-6">
            <div className="space-y-3">
              <h3 className="font-medium text-green-600">‚úÖ Do</h3>
              <ul className="space-y-2 text-sm">
                <li>
                  ‚Ä¢ Use <code>variant=&quot;destructive&quot;</code> for dangerous actions
                </li>
                <li>
                  ‚Ä¢ Use <code>size=&quot;lg&quot;</code> for primary CTAs
                </li>
                <li>‚Ä¢ Provide descriptive button text</li>
                <li>
                  ‚Ä¢ Use <code>disabled</code> state during loading
                </li>
                <li>
                  ‚Ä¢ Use <code>asChild</code> for navigation links
                </li>
              </ul>
            </div>
            <div className="space-y-3">
              <h3 className="font-medium text-red-600">‚ùå Don&apos;t</h3>
              <ul className="space-y-2 text-sm">
                <li>‚Ä¢ Don&apos;t use too many different variants in one area</li>
                <li>‚Ä¢ Don&apos;t make buttons too small for mobile</li>
                <li>
                  ‚Ä¢ Don&apos;t use <code>link</code> variant for actions
                </li>
                <li>‚Ä¢ Don&apos;t forget focus states for accessibility</li>
                <li>
                  ‚Ä¢ Don&apos;t overuse <code>brand</code> variant
                </li>
              </ul>
            </div>
          </div>
        </section>
      </div>
    </div>
  );
}

export default ButtonExamples;
