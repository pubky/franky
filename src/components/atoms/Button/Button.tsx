'use client';

import * as React from 'react';
import { Slot } from '@radix-ui/react-slot';
import { cva, type VariantProps } from 'class-variance-authority';

import { cn } from '@/libs';

const buttonVariants = cva(
  "inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-semibold cursor-pointer rounded-full",
  {
    variants: {
      variant: {
        default: 'bg-brand/20 text-brand border shadow-xs hover:!bg-brand/30 border-brand',
        destructive:
          'bg-destructive text-white shadow-xs hover:bg-destructive/90 focus-visible:ring-destructive/20 dark:focus-visible:ring-destructive/40 dark:bg-destructive/60',
        outline:
          'border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50',
        secondary: 'bg-secondary text-secondary-foreground shadow-xs hover:bg-secondary/80',
        ghost: 'hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50',
        brand: 'bg-brand text-background border-brand shadow-xs hover:bg-brand/90',
        link: 'text-primary underline-offset-4 hover:underline',
      },
      size: {
        default: 'h-10 gap-1 px-4 py-2 has-[>svg]:px-4',
        sm: 'h-8 gap-1.5 px-3 has-[>svg]:px-2.5',
        lg: 'h-10 gap-1 px-8 py-8 has-[>svg]:px-8',
        icon: 'size-9',
      },
    },
    defaultVariants: {
      variant: 'default',
      size: 'default',
    },
  },
);

function Button({
  className,
  variant,
  size,
  asChild = false,
  ...props
}: React.ComponentProps<'button'> &
  VariantProps<typeof buttonVariants> & {
    asChild?: boolean;
  }) {
  const Comp = asChild ? Slot : 'button';

  const getTestId = () => {
    const effectiveVariant = variant || 'default'; // Handle undefined variant
    if (effectiveVariant === 'ghost' && size === 'icon') return 'popover-button';
    if (effectiveVariant === 'ghost') return 'button-ghost';
    if (effectiveVariant === 'outline') return 'button-outline';
    if (effectiveVariant === 'secondary') return 'button-secondary';
    if (effectiveVariant === 'default') return 'button-default';
    return 'button';
  };

  return (
    <Comp
      data-slot="button"
      data-testid={getTestId()}
      data-variant={variant}
      data-size={size}
      className={cn(buttonVariants({ variant, size, className }))}
      {...props}
    />
  );
}

export { Button, buttonVariants };
