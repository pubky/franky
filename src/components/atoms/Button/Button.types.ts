import * as React from 'react';

export type ButtonVariant =
  | 'default'
  | 'destructive'
  | 'outline'
  | 'secondary'
  | 'ghost'
  | 'brand'
  | 'link'
  | 'dark'
  | 'dark-outline';

export type ButtonSize = 'default' | 'sm' | 'lg' | 'icon';

export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  className?: React.HTMLAttributes<HTMLButtonElement>['className'];
  variant?: ButtonVariant;
  size?: ButtonSize;
  asChild?: boolean;
}
