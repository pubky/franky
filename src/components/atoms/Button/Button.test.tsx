import { describe, it, expect, vi } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from './Button';

describe('Button', () => {
  it('renders with default props', () => {
    render(<Button>Click me</Button>);
    const button = screen.getByRole('button', { name: /click me/i });
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('bg-brand/20', 'text-brand');
  });

  it('renders different variants correctly', () => {
    const { rerender } = render(<Button variant="secondary">Secondary</Button>);
    let button = screen.getByRole('button');
    expect(button).toHaveClass('bg-secondary', 'text-secondary-foreground');

    rerender(<Button variant="outline">Outline</Button>);
    button = screen.getByRole('button');
    expect(button).toHaveClass('border', 'bg-background');

    rerender(<Button variant="ghost">Ghost</Button>);
    button = screen.getByRole('button');
    expect(button).toHaveClass('hover:bg-accent');

    rerender(<Button variant="brand">Brand</Button>);
    button = screen.getByRole('button');
    expect(button).toHaveClass('bg-brand', 'text-background');
  });

  it('renders different sizes correctly', () => {
    const { rerender } = render(<Button size="sm">Small</Button>);
    let button = screen.getByRole('button');
    expect(button).toHaveClass('h-8', 'gap-1.5', 'px-3');

    rerender(<Button size="lg">Large</Button>);
    button = screen.getByRole('button');
    expect(button).toHaveClass('h-10', 'px-8');

    rerender(<Button size="icon">Icon</Button>);
    button = screen.getByRole('button');
    expect(button).toHaveClass('size-9');
  });

  it('handles click events', () => {
    const handleClick = vi.fn();
    render(<Button onClick={handleClick}>Click me</Button>);

    fireEvent.click(screen.getByRole('button'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('can be disabled', () => {
    render(<Button disabled>Disabled</Button>);
    const button = screen.getByRole('button');
    expect(button).toBeDisabled();
    expect(button).toHaveClass('disabled:pointer-events-none');
  });

  it('renders as child component when asChild is true', () => {
    render(
      <Button asChild>
        <a href="/test">Link Button</a>
      </Button>,
    );

    const link = screen.getByRole('link', { name: /link button/i });
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', '/test');
  });

  it('applies custom className', () => {
    render(<Button className="custom-class">Custom</Button>);
    const button = screen.getByRole('button');
    expect(button).toHaveClass('custom-class');
  });
});
