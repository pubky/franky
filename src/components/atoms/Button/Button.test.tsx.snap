// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button - Snapshots > matches snapshot for asChild prop 1`] = `
<a
  class="inline-flex items-center justify-center whitespace-nowrap text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*=\\"size-\\"])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-semibold cursor-pointer rounded-full border shadow-xs bg-brand/16 text-brand hover:!bg-brand/30 border-brand h-10 gap-1 px-4 py-2 has-[>svg]:px-4"
  data-slot="button"
  href="/test"
>
  Link Button
</a>
`;

exports[`Button - Snapshots > matches snapshot for brand variant 1`] = `
<button
  class="inline-flex items-center justify-center whitespace-nowrap text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*=\\"size-\\"])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-semibold cursor-pointer rounded-full border shadow-xs bg-brand text-background border-brand hover:bg-brand/90 h-10 gap-1 px-4 py-2 has-[>svg]:px-4"
  data-slot="button"
  data-variant="brand"
>
  Brand
</button>
`;

exports[`Button - Snapshots > matches snapshot for default size 1`] = `
<button
  class="inline-flex items-center justify-center whitespace-nowrap text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*=\\"size-\\"])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-semibold cursor-pointer rounded-full border shadow-xs bg-brand/16 text-brand hover:!bg-brand/30 border-brand h-10 gap-1 px-4 py-2 has-[>svg]:px-4"
  data-slot="button"
>
  Default Size
</button>
`;

exports[`Button - Snapshots > matches snapshot for default variant 1`] = `
<button
  class="inline-flex items-center justify-center whitespace-nowrap text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*=\\"size-\\"])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-semibold cursor-pointer rounded-full border shadow-xs bg-brand/16 text-brand hover:!bg-brand/30 border-brand h-10 gap-1 px-4 py-2 has-[>svg]:px-4"
  data-slot="button"
>
  Default
</button>
`;

exports[`Button - Snapshots > matches snapshot for disabled state 1`] = `
<button
  class="inline-flex items-center justify-center whitespace-nowrap text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*=\\"size-\\"])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-semibold cursor-pointer rounded-full border shadow-xs bg-brand/16 text-brand hover:!bg-brand/30 border-brand h-10 gap-1 px-4 py-2 has-[>svg]:px-4"
  data-slot="button"
  disabled=""
>
  Disabled
</button>
`;

exports[`Button - Snapshots > matches snapshot for ghost variant 1`] = `
<button
  class="inline-flex items-center justify-center whitespace-nowrap text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*=\\"size-\\"])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-semibold cursor-pointer rounded-full border shadow-xs hover:text-accent-foreground hover:bg-accent/50 border-none h-10 gap-1 px-4 py-2 has-[>svg]:px-4"
  data-slot="button"
  data-variant="ghost"
>
  Ghost
</button>
`;

exports[`Button - Snapshots > matches snapshot for icon size 1`] = `
<button
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*=\\"size-\\"])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-semibold cursor-pointer rounded-full border shadow-xs bg-brand/16 text-brand hover:!bg-brand/30 border-brand size-9"
  data-size="icon"
  data-slot="button"
>
  üîç
</button>
`;

exports[`Button - Snapshots > matches snapshot for large size 1`] = `
<button
  class="inline-flex items-center justify-center whitespace-nowrap transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*=\\"size-\\"])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/40 aria-invalid:border-destructive cursor-pointer rounded-full border shadow-xs bg-brand/16 text-brand hover:!bg-brand/30 border-brand h-auto gap-2 px-2.5 py-4 sm:px-8 sm:py-5 text-sm font-bold leading-normal sm:flex-1"
  data-size="lg"
  data-slot="button"
>
  Large
</button>
`;

exports[`Button - Snapshots > matches snapshot for outline variant 1`] = `
<button
  class="inline-flex items-center justify-center whitespace-nowrap text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*=\\"size-\\"])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-semibold cursor-pointer rounded-full border shadow-xs hover:text-accent-foreground bg-input/30 border-input hover:bg-input/50 h-10 gap-1 px-4 py-2 has-[>svg]:px-4"
  data-slot="button"
  data-variant="outline"
>
  Outline
</button>
`;

exports[`Button - Snapshots > matches snapshot for secondary variant 1`] = `
<button
  class="inline-flex items-center justify-center whitespace-nowrap text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*=\\"size-\\"])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-semibold cursor-pointer rounded-full border shadow-xs bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 gap-1 px-4 py-2 has-[>svg]:px-4"
  data-slot="button"
  data-variant="secondary"
>
  Secondary
</button>
`;

exports[`Button - Snapshots > matches snapshot for small size 1`] = `
<button
  class="inline-flex items-center justify-center whitespace-nowrap text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*=\\"size-\\"])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-semibold cursor-pointer rounded-full border shadow-xs bg-brand/16 text-brand hover:!bg-brand/30 border-brand h-8 gap-1.5 px-3 has-[>svg]:px-2.5"
  data-size="sm"
  data-slot="button"
>
  Small
</button>
`;

exports[`Button - Snapshots > matches snapshot with custom className 1`] = `
<button
  class="inline-flex items-center justify-center whitespace-nowrap text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*=\\"size-\\"])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-semibold cursor-pointer rounded-full border shadow-xs bg-brand/16 text-brand hover:!bg-brand/30 border-brand h-10 gap-1 px-4 py-2 has-[>svg]:px-4 custom-class"
  data-slot="button"
>
  Custom
</button>
`;

exports[`Button - Snapshots > matches snapshot with default props 1`] = `
<button
  class="inline-flex items-center justify-center whitespace-nowrap text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*=\\"size-\\"])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-semibold cursor-pointer rounded-full border shadow-xs bg-brand/16 text-brand hover:!bg-brand/30 border-brand h-10 gap-1 px-4 py-2 has-[>svg]:px-4"
  data-slot="button"
>
  Default Button
</button>
`;
